Missing includes

The following minimal example `main.cpp`:

``` C++
#include <array>
int main( void ) { return 0; };
```

compiled with

```
g++ -std=c++11 -I ../include/ -I ../include/foobar/ main.cpp
```

will result in the following errors:

```
In file included from ../include/foobar/FFT_Properties.hpp:3:0,
                 from ../include/foobar/FFT.hpp:3,
                 from main.cpp:7:
../include/foobar/FFT_Kind.hpp:21:46: error: expected class-name before ‘{’ token
         struct IsComplexInput: std::true_type{};
                                              ^
../include/foobar/FFT_Kind.hpp:24:73: error: expected class-name before ‘{’ token
         struct IsComplexInput< FFT_Kind::Real2Complex >: std::false_type{};
                                                                         ^
../include/foobar/FFT_Kind.hpp:30:47: error: expected class-name before ‘{’ token
         struct IsComplexOutput: std::true_type{};
                                               ^
../include/foobar/FFT_Kind.hpp:33:73: error: expected class-name before ‘{’ token
         struct IsComplexOutput< FFT_Kind::Complex2Real>: std::false_type{};
                                                                         ^
In file included from ../include/foobar/FFT_Definition.hpp:6:0,
                 from ../include/foobar/FFT.hpp:6,
                 from main.cpp:7:
../include/foobar/util.hpp:73:27: error: ‘size_t’ has not been declared
     template< typename T, size_t N >
                           ^
```

It seems like they forgot to include `type_traits` and `utility` themselves. Adding `#include <array>` before the include of `FFT.hpp` will solve those errors, because `array` includes many of those standard needed libraries.

--- a/include/foobar/FFT_Kind.hpp
+++ b/include/foobar/FFT_Kind.hpp
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <type_traits>
+
 namespace foobar {
 
     /**

--- a/include/foobar/util.hpp
+++ b/include/foobar/util.hpp
@@ -70,7 +70,7 @@ namespace foobar {
         using type = AddConstVal_t<T> [];
     };
 
-    template< typename T, size_t N >
+    template< typename T, std::size_t N >
     struct AddConstVal<T [N]>
     {
         using type = AddConstVal_t<T> [N];

